react-and-redux-style-guide/README.md

```diff
@@ -0,0 +1,469 @@
+# RingCentral React and Redux Style Guide
+
+- [Introduction](#introduction)
+- [Code structure](#code-structure)
+- [Naming](#naming)
+- [Good practices](#good-practices)
+
+# Introduction
+
+This is a set of recommended patterns and best practices for writing applications using React+Redux.
+
+# Code structure
+
+## # Put all Redux codes in `redux` directory inside your module 
+> Reason: this helps to keep module isolated
+
+## # Keep reducers clean by moving logic to helpers
+> Reason: a lot of logic inside the reducer complicates an understanding of data changes produced by the action
+```typescript
+// BAD
+export const someReducer: Reducer<SomeStore> = (state: SomeStore = initialState, action: ActionTypes) => {
+    switch (action.type) {
+            case UPDATE: {
+                let oldItems = state.items;
+                let someData = getSomeDraftData(oldItems).map(() => {
+                    ...
+                });       
+                ...
+                let items = someData.reduce((acc, item) => {
+                    ...
+                }, {});
+                return {
+                    ...state,
+                    items,
+                };
+            }
+            default: {
+                return state;
+            }
+    }
+};
+```
+```typescript
+// GOOD
+export const someReducer: Reducer<SomeStore, ActionTypes> = (state = initialState, action) => {
+    switch (action.type) {
+            case UPDATE: {
+                return {
+                    ...state,
+                    items: calculateItemsUpdates(state.items),
+                };
+            }
+            default: {
+                return state;
+            }
+    }
+};
+``` 
+ 
+## # Put Redux connection logic to Containers
+> Reason: this makes the connection between a React component and Redux storage more transparent and isolated.
+```typescript
+// EXAMPLE
+// /someModule/containers/FilterContainer.tsx
+import {Filter} from '../components/settings/Filter';
+import {itemsSelector} from '../redux/selectors';
+import {updateFilter} from '../redux/actions';
+
+const mapStateToProps = (state) => ({
+    items: itemsSelector(state),
+});
+
+const mapDispatchToProps = {
+    onChange: updateFilter,
+};
+
+export const FilterContainer: React.FC<{}> = connect(
+    mapStateToProps,
+    mapDispatchToProps,
+)(Filter);
```

Comment:
Prefer using functional components and `useSelector`/`useDispatch`.

Reply:
Discussion about class and functional components looks out of this SG scope. I agree with you in general, but class components are still a big part of our codebase, so we need to have some guidelines for them